src/openms/source/FORMAT/PepXMLFile.cpp

```diff
@@ -281,21 +281,24 @@ namespace OpenMS
         AASequence seq = h.getSequence();
         double precursor_neutral_mass = seq.getMonoWeight();
 
-        Int scan_index;
+        int scan_index = count;
+        int scan_nr = 0;
+        
         if (lookup.empty())
         {
           if (it->metaValueExists("RT_index")) // Setting metaValue "RT_index" in XTandemXMLFile in the case of X! Tandem.
           {
-            scan_index = it->getMetaValue("RT_index");
-          }
-          else
-          {
-            scan_index = count;
+            scan_index = it->getMetaValue("RT_index");            
           }
+          scan_nr = scan_index + 1;          
```

Comment:
Another option which I prefer would be: if the spectrum lookup is not possible, start the scan_index with 0 (count = 0), and set scan_number = scan_index + 1. This should be closest to the original pattern where scan_index starts with 0 and scan_nr with 1.